src/api/services/iuf/sessions.go

```diff
@@ -354,78 +354,82 @@ func (s iufService) ProcessOutput(session *iuf.Session, workflow *v1alpha1.Workf
-					activity.OperationOutputs["stage_params"].(map[string]interface{})["process-media"].(map[string]interface{})["products"].(map[string]interface{})[fmt.Sprintf("%v", manifest["name"])].(map[string]interface{})["parent_directory"] = nodeStatus.Outputs.Parameters[1].Value.String()
-				}
-			}
-			session.Products = activity.Products
-			// update activity
-			configmap, err := s.iufObjectToConfigMapData(activity, activity.Name, LABEL_ACTIVITY)
+			err := s.processOutputOfProcessMedia(&activity, workflow)
 			if err != nil {
 				s.logger.Error(err)
 				return err
 			}
-
-			_, err = s.k8sRestClientSet.
-				CoreV1().
-				ConfigMaps(DEFAULT_NAMESPACE).
-				Update(
-					context.TODO(),
-					&configmap,
-					v1.UpdateOptions{},
-				)
+			session.Products = activity.Products
+			// update activity
+			_, err = s.updateActivity(activity)
 			if err != nil {
 				s.logger.Error(err)
 				return err
 			}
 			return nil
+		} else {
+			for _, task := range tasks {
+				operationName := task.TemplateRef.Name
+				nodeStatus := workflow.Status.Nodes.FindByDisplayName(task.Name)
+				s.logger.Infof("process output of: %s, %v", operationName, nodeStatus.Outputs)
+				s.updateActivityOperationOutputFromWorkflow(activity, *session, nodeStatus, operationName, "")
+			}
+			return nil
 		}
-		for _, task := range tasks {
-			operationName := task.TemplateRef.Name
-			nodeStatus := workflow.Status.Nodes.FindByDisplayName(task.Name)
-			s.logger.Infof("process output of: %s, %v", operationName, nodeStatus.Outputs)
-			s.updateActivityOperationOutputFromWorkflow(activity, *session, nodeStatus, operationName, "")
-		}
-		return nil
 	default:
 		return fmt.Errorf("stage_type: %s is not supported", workflow.Labels["stage_type"])
 	}
 
 }
 
+func (s iufService) processOutputOfProcessMedia(activity *iuf.Activity, workflow *v1alpha1.Workflow) error {
+	nodesWithOutputs := workflow.Status.Nodes.Filter(func(nodeStatus v1alpha1.NodeStatus) bool {
+		return nodeStatus.Outputs.HasOutputs() && len(nodeStatus.Outputs.Parameters) == 2
+	})
+	if len(nodesWithOutputs) == 0 {
+		return nil
+	}
+	activity.OperationOutputs = map[string]interface{}{
+		"stage_params": map[string]interface{}{
+			"process-media": map[string]interface{}{
+				"products": map[string]interface{}{},
+			},
+		},
+	}
+	activity.Products = []iuf.Product{}
+	for _, nodeStatus := range nodesWithOutputs {
+		var manifest map[string]interface{}
+		err := yaml.Unmarshal([]byte(nodeStatus.Outputs.Parameters[0].Value.String()), &manifest)
+		if err != nil {
+			s.logger.Error(err)
+			return err
+		}
+		// validate iuf product manifest
+		data, _ := yaml.Marshal(manifest)
+		validated := true
+		err = iuf.Validate(data)
+		if err != nil {
+			s.logger.Error(err)
+			validated = false
+		}
+		jsonManifest, _ := json.Marshal(manifest)
+		if manifest["name"] != nil {
```

Comment:
this really goes back to: if a manifest is invalid at line 413, should we just fail this whole thing?

Reply:
Discussed with Di. We think it's best to omit any products that do not have a valid manifest, so that subsequent stages don't fail trying to process those products. We will make a best effort to log the name, version, and tar file path above in line 412 and just note that it has been skipped. Then the admin can recognize and fix the problem.

Reply:
Filed a Jira: https://jira-pro.its.hpecorp.net:8443/browse/CASM-3696